src/beneuro_data/conversion/anipose_interface.py

```diff
@@ -73,25 +85,29 @@ def add_to_nwbfile(
         nwbfile: NWBFile,
         metadata: Optional[DeepDict] = None,
         stub_test: bool = False,
+        use_default_fps: bool = True,
     ):
-        try:
+        # Alignment: As cameras start recording when PyControl sends them the signal at t=0,
+        # and in theory sends a signal with DEFAULT_FPS frequency, set the default option for the
+        # timing of the frames to use `starting_time` and `rate` instead of explicit timestamps.
+        if use_default_fps:
+            timestamps = None
+            starting_time = 0.0
+            rate = float(self.DEFAULT_FPS)
+
+        elif not use_default_fps:
             timestamps = self.get_original_timestamps()
             starting_time = None
             rate = None
-        except:
-            # if we can't get the timestamps from spikeglx for whatever reason
-            # then just assume that the frames come with DEFAULT_FPS
-            # and that the first one is at t=0 which should be true
-            timestamps = None
-            starting_time = 0.0
-            rate = float(DEFAULT_FPS)
 
         keypoint_series_objects = []
-        for i, keypoint_name in enumerate(self.keypoint_names):
+        for keypoint_name in self.keypoint_names:
             keypoint_series = PoseEstimationSeries(
                 name=keypoint_name,
                 description=f"Marker placed at {keypoint_name.replace('_', ' ')}",
-                data=self.pose_data[:, i, :],
+                data=self.pose_data[
+                    [f"{keypoint_name}_x", f"{keypoint_name}_y", f"{keypoint_name}_z"]
+                ].to_numpy(),
                 unit="a.u.",  # TODO
                 reference_frame="(0, 0, 0) is what?",  # TODO
```

Comment:
@martinesparza @gen21kawa can you guys fill these? We might not have a unit, but the reference/origin would be useful to write here.

Reply:
Same for the angles.

Reply:
We're using hip_center as a reference point (0,0,0) for triangulation, so I guess that the reference frame (0, 0, 0) would be the position of hip_center.(The position of hip_center would probably change throughout the recording) The unit is supposed to be in mm but the values do not seem to be in mm...

Here is the explanation on the reference frame from Anipose.
> In order to properly compare across different trials, different animals, and different setups, it may be useful to standardize 3D coordinates relative to a common reference frame.

Anipose allows configuration of this by specifying 2 sets of points to use as axes, and which axes these should be.

The algorithm to determine the axes is as follows: - the first axis is taken as given - the second axis is orthogonalized with respect to the first - the third axis is the cross product of the first two axes

An axis is specified as a pair of points, with the axis going from the first to the second point.

Furthermore, it is often useful to set the zero to a standard reference point. Anipose allows this too.

Our configuration looks like this:

[triangulation]
axes = [
    ["right_shoulder", "shoulder_center","left_shoulder"],
    ["shoulder_center", "hip_center"]
]
reference_point = "hip_center"


The angles are defined in the config file for triangulation. We're computing the following six angles defined as:
[angles]
left_elbow_angle = ["left_shoulder", "left_elbow", "left_wrist"]
right_elbow_angle = ["right_shoulder", "right_elbow", "right_wrist"]
left_knee_angle = ["hip_center","left_knee","left_ankle"]
right_knee_angle = ["hip_center","right_knee","right_ankle"]
left_ankle_angle = ["left_knee","left_ankle","left_foot"]
right_ankle_angle = ["right_knee","right_ankle","right_foot"]

The unit of the angle is in degree (0-360 or 0-180). For now, we are using 2d values for each angle for convenience though the second dimension is always set to 0 and does not mean anything. So the reference frame (0,0) would be when the angle is 0 (the three keypoints for the angle is in the straight line). 

Please check this [page](https://github.com/lambdaloop/anipose/blob/b9859b5e575e962581920d36807f2987da17ebfc/docs/sphinx/start3d.rst#L200) from Anipose for more info.

Reply:
I will wait for @gen21kawa to confirm if we need to substract hip_center position. Otherwise, `units == "mm"` (theoretically) and `reference = hip_center`

Reply:
We don't need to subtract the coordinates of hip_center by ourselves. I was a bit confused too, so I checked the triangulation.py in Anipose. They compute the median of the reference point (hip_center for us) across all the frames, and subtract the median from all the points. This is why the xyz coordinates of hip_center are not always (0,0,0).